pkgs/desktops/gnome/core/gdm/default.nix

```diff
@@ -193,5 +193,6 @@ stdenv.mkDerivation (finalAttrs: {
     license = licenses.gpl2Plus;
     maintainers = teams.gnome.members;
     platforms = platforms.linux;
+    mainProgram = "gdm";
```

Comment:
The executable is not really be supposed to run by end user e.g. with `nix run` so I don’t see much of a point in this this.

Reply:
I can see where you're coming from but `nix run` is not the only user of `meta.mainProgram` and there might be future users too. This is just metadata recognising `gdm` as the "primary" executable that the package provides and does not imply a recommendation to run it via `nix run`.

Reply:
I do not think it makes sense to have a primary executable in non-user facing scenarios in the first place. There, you want to be precise what you are calling, not relying on an indirection.